src/net/sourceforge/kolmafia/AdventureResult.java

```diff
@@ -857,20 +851,18 @@ public String toConditionString() {
    */
   @Override
   public boolean equals(final Object o) {
-    if (!(o instanceof AdventureResult)) {
+    if (!(o instanceof AdventureResult ar)) {
```

Comment:
I am not familiar with this. Which Java version did this come in?

Reply:
This is [pattern matching for instanceof](https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof-operator.html): previewed in 14, released properly in 16.

Avoids a manual cast, so very nice in my book.